### PR TITLE
Low-stock alert configuration in the Product Page

### PR DESCRIPTION
**Pull request by aleeks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Low-stock alert configuration in the Product Page
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/FF-121 <br> http://forge.prestashop.com/browse/FF-122 <br> http://forge.prestashop.com/browse/FF-123
| How to test?  | Check tickets.

Please, test all case: 
- simple product: with value, empty, and 0 value,
- combination: with value, empty, and 0 value,
- bulk actions: with value, and 0 (empty keep the original value so...)
=> With the bulk actions, please check after if your value is good

### CODE REVIEW

**Comment by aleeks** [comment]
See with ?w=1

**Comment by aleeks** [comment]
ping @eternoendless @marionf @maximebiloe @Quetzacoalt91

**Comment by Quetzacoalt91** [comment]
Can you please rebase? :grimacing:

File: src/PrestaShopBundle/Form/Admin/Product/ProductCombination.php [hunk 1]
```diff
@@ -140,6 +140,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 new Assert\Type(array('type' => 'numeric')),
             )
         ))
+        ->add('attribute_low_stock', 'Symfony\Component\Form\Extension\Core\Type\NumberType', array(
+            'label' => $this->translator->trans('Low stock level', [], 'Admin.Catalog.Feature'),
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Form/Admin/Product/ProductCombination.php]: Son, you disappoint me.

`[]` => `array()`

Comment (aleeks) [reply src/PrestaShopBundle/Form/Admin/Product/ProductCombination.php]: I just copy and past... f*************k!!

**Comment by Quetzacoalt91** [comment]
So, this PR just adds the fields in the forms to manage the low level alert?

**Comment by aleeks** [comment]
Yes @Quetzacoalt91 
- Add new field in base,
- Add field in model
- Add field in form & update it

File: classes/Combination.php [hunk 2]
```diff
@@ -55,6 +55,8 @@ class CombinationCore extends ObjectModel
 
     public $minimal_quantity = 1;
 
+    public $low_stock = null;
```
Comment (eternoendless) [inline classes/Combination.php]: Please don't forget to add PHPDoc for new attributes.

File: classes/Combination.php [hunk 3]
```diff
@@ -71,21 +73,22 @@ class CombinationCore extends ObjectModel
...
+            'wholesale_price' =>   array('type' => self::TYPE_FLOAT, 'shop' => true, 'validate' => 'isPrice', 'size' => 27),
+            'price' =>             array('type' => self::TYPE_FLOAT, 'shop' => true, 'validate' => 'isNegativePrice', 'size' => 20),
             'ecotax' =>            array('type' => self::TYPE_FLOAT, 'shop' => true, 'validate' => 'isPrice', 'size' => 20),
             'weight' =>            array('type' => self::TYPE_FLOAT, 'shop' => true, 'validate' => 'isFloat'),
-            'unit_price_impact' =>    array('type' => self::TYPE_FLOAT, 'shop' => true, 'validate' => 'isNegativePrice', 'size' => 20),
-            'minimal_quantity' =>    array('type' => self::TYPE_INT, 'shop' => true, 'validate' => 'isUnsignedId', 'required' => true),
+            'unit_price_impact' => array('type' => self::TYPE_FLOAT, 'shop' => true, 'validate' => 'isNegativePrice', 'size' => 20),
+            'minimal_quantity' =>  array('type' => self::TYPE_INT, 'shop' => true, 'validate' => 'isUnsignedId', 'required' => true),
+            'low_stock' =>         array('type' => self::TYPE_INT, 'shop' => true, 'allow_null' => true, 'validate' => 'isInt'),
```
Comment (eternoendless) [inline classes/Combination.php]: Shouldn't `'validate' => 'isInt'` be `isUnsignedInt` ?

File: install-dev/data/db_structure.sql [hunk 4]
```diff
@@ -1400,6 +1400,7 @@ CREATE TABLE `PREFIX_product` (
   `ecotax` decimal(17,6) NOT NULL DEFAULT '0.00',
   `quantity` int(10) NOT NULL DEFAULT '0',
   `minimal_quantity` int(10) unsigned NOT NULL DEFAULT '1',
+  `low_stock` int(10) NULL DEFAULT NULL,
```
Comment (eternoendless) [inline install-dev/data/db_structure.sql]: Shouldn't this be unsigned?

File: classes/Combination.php (hunk 2 above)
Comment (eternoendless) [reply classes/Combination.php]: Could this be changed everywhere to "low_stock_threshold" ?

File: src/PrestaShopBundle/Resources/views/Admin/Product/Include/form_combination.html.twig [hunk 5]
```diff
@@ -101,6 +101,18 @@
                     </div>
                 </div>
                 <h2 class="title">
+                  {{ 'Stock alerts'|trans({}, 'Admin.Catalog.Feature') }}
+                </h2>
+                <div class="row">
+                  <div class="col-md-3">
+                    <fieldset class="form-group">
+                      <label class="form-control-label">{{ form.attribute_low_stock.vars.label|raw }}</label>
```
Comment (eternoendless) [inline src/PrestaShopBundle/Resources/views/Admin/Product/Include/form_combination.html.twig]: Why the "|raw" filter?

**Review by eternoendless — CHANGES_REQUESTED** [review]

File: install-dev/data/db_structure.sql (hunk 4 above)
Comment (aleeks) [reply install-dev/data/db_structure.sql]: Nope, allow negative number

File: classes/Combination.php (hunk 3 above)
Comment (aleeks) [reply classes/Combination.php]: Nope, allow negative number

File: js/admin/products.js [hunk 6]
```diff
@@ -318,7 +319,8 @@ product_tabs['Combinations'] = new function(){
 						eco_tax,
 						upc,
 						minimal_quantity,
-						available_date
+						available_date,
+            low_stock_threshold
```
Comment (eternoendless) [inline js/admin/products.js]: Indentation problem here 😬

**Review by eternoendless — CHANGES_REQUESTED** [review]
Just a small thing left!

File: js/admin/products.js (hunk 6 above)
Comment (aleeks) [reply js/admin/products.js]: done!

**Review by eternoendless — APPROVED** [review]
Code looks good!

**Comment by eternoendless** [comment]
Thank you @aleeks

**Pull request merged**